test/runtests.jl

```diff
@@ -34,3 +34,10 @@ println("Testing miscellaneous functionality...")
 tic()
 include("MiscTest.jl")
 println("done (took $(toq()) seconds).")
+
+if Base.JLOptions().opt_level >= 3 && VERSION >= v"0.5"
```

Comment:
Could try to run the build sysimg.jl script first in .travis.yml file and then start the actual tests?